applications/serial_lte_modem/src/gpio/slm_at_gpio.c

```diff
@@ -0,0 +1,312 @@
+					rsp_send(rsp_buf, strlen(rsp_buf));
+				} else if (op == SLM_GPIO_OP_TOGGLE) {
+					LOG_DBG("Toggle pin: %d", cur->pin);
+					ret = gpio_pin_toggle(gpio_dev, pin);
+					if (ret < 0) {
+						LOG_ERR("Cannot toggle gpio");
+						return ret;
+					}
+				}
+			}
+		}
+	}
+
+	return 0;
+}
+
+/**@brief handle AT#XGPIOCFG commands
+ *  AT#XGPIOCFG=<op>,<pin>
+ *  AT#XGPIOCFG?
+ *  AT#XGPIOCFG=?
+ */
+int handle_at_gpio_configure(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t pin = 0xff, op = 0xff;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (at_params_valid_count_get(&at_param_list) == 0) {
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			LOG_ERR("Fail to get op: %d", err);
+			return err;
+		}
+		err = at_params_short_get(&at_param_list, 2, &pin);
+		if (err < 0) {
+			LOG_ERR("Fail to get pin: %d", err);
+			return err;
+		}
+		err = do_gpio_pin_configure_set(op, (gpio_pin_t)pin);
+		break;
+	case AT_CMD_TYPE_READ_COMMAND:
+		err = do_gpio_pin_configure_read();
+		break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief handle AT#XGPIOCFG commands
+ *  AT#XGPIO=<pin>,<op>[,<value>]
+ *  AT#XGPIO? READ command not supported
+ *  AT#XGPIO=?
+ */
+int handle_at_gpio_operate(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t pin = 0xff, op = 0xff, value = 0xff;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (at_params_valid_count_get(&at_param_list) == 0) {
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			LOG_ERR("Fail to get op: %d", err);
+			return err;
+		}		
+		if (op > SLM_GPIO_OP_TOGGLE) {
+			LOG_ERR("Fail to operate gpio: %d", op);
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 2, &pin);
+		if (err < 0) {
+			LOG_ERR("Fail to get pin: %d", err);
+			return err;
+		}
+		if (pin > MAX_GPIO_PIN) {
+			LOG_ERR("Incorrect <pin>: %d", pin);
+			return -EINVAL;
+		}
+		if (op == SLM_GPIO_OP_WRITE) {
+			if (at_params_valid_count_get(&at_param_list) == 4) {
+				err = at_params_short_get(&at_param_list, 3, &value);
+				if (err < 0) {
+					LOG_ERR("Fail to get value: %d", err);
+					return err;
+				}
+				if (value != 1 && value != 0) {
+					LOG_ERR("Fail to set gpio value: %d", value);
+					return -EINVAL;
+				}
+			}
+			else {
+				LOG_ERR("Fail to get value: %d", value);
+				return -EINVAL;
+			}
+		}
+		err = do_gpio_pin_operate(op, (gpio_pin_t)pin, value);
+		break;
+	case AT_CMD_TYPE_READ_COMMAND:
+		break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		break;
+
+	default:
+		break;
+	}
+	return err;
+}
+
+int slm_at_gpio_init(void)
+{
+	int err = 0;
+
+	gpio_dev = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
+	if (gpio_dev == NULL) {
+		LOG_ERR("GPIO_0 bind error");
+		return -EIO;
+	}
+
+	return err;
+}
+
+int slm_at_gpio_uninit(void)
+{
+	int err = 0;
```

Comment:
Plan to do something here?
Otherwise just return 0

Reply:
return 0